optimum/onnxruntime/utils.py

```diff
@@ -224,7 +224,8 @@ def check_io_binding(providers: List[str], use_io_binding: Optional[bool] = None
                 "No need to enable IO Binding if the provider used is not CUDAExecutionProvider. IO Binding will be turned off."
             )
         use_io_binding = False
-
+    elif use_io_binding is None:
```

Comment:
About your first question, I do not think so because in the `to` method we update the providers, which can be blocking to have `use_io_binding`. But we actually dont want to use io binding when the initialization was done so that `self._use_io_binding = False`.

Reply:
I think you're right, removed it.